src/routeObject.js

```diff
@@ -1,6 +1,59 @@
 import React from 'react';
 import {Route, IndexRoute} from 'react-router';
+
 import App from './pages/App';
+import NotFound from './pages/PageNotFound';
+import store from './store';
+
+const lessons = store.getState().lessons;
+const courses = store.getState().context['courseContext'].keys();
+
+const lessonArray = Object.keys(lessons).map((key) => lessons[key]['path']);
+const courseArray = courses.map((course) => course.slice(1).replace(/\/index.md/i, ''));
+
+const validPathTest = (lesson, path) => {
+  if(lesson){
+    return (lessonArray.indexOf(path) > -1);
+  }else{
+    return (courseArray.indexOf(path) > -1);
+  }
+};
+
+const pathTest = (nextState, replace, callback) => {
+  const params = nextState.params;
+  let path = nextState.location.pathname;
+  if(path.indexOf('/') !== 0){
+    path = '/' + path;
+  }
+  if(path.lastIndexOf('/') === path.length-1){
+    path = path.slice(0, -1);
+  }
+  
+  const pathCorrect = validPathTest(params.lesson, path);
+
+  if(!pathCorrect){
+    replace({pathname:'/PageNotFound', state: path});
+  }
+  callback();
+};
+
+const saveURL = (nextState, replace, callback) => {
+  const path = nextState.location.state;
```

Comment:
Og, kunne vi lagt på en kommentar her, noe a la: "We are using window.history directly, since we could not find a way to use the replace-function we receive from onEnter (react router 2) to change the URL of the browser without redirecting. The purpose of using replaceState is just to show/leave the URL the user typed into the browser even after we have redirected to PageNotFound."

Gjør gjerne endringer eller tillegg til kommentaren slik at den blir riktig og forståelig når noen senere kommer og har tenkt å prøve å gjøre om på mekanismene her.

Legg også gjerne til en kommentar til serverSideRedirectCheck, og forklar hva den brukes til.